### PR TITLE
Move with gyro correct

### PR DESCRIPTION
**Pull request by JuliaLupaczyk**
Added gyro stuff

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/drivetrain/MockDriveTrain.java [hunk 1]
```diff
@@ -33,4 +33,13 @@ public double getEncoderPositionInches() {
   public double getEncoderVelocityInches() {
     return 0;
   }
+
+  public void moveUsingGyro(double forwardBackward, double rotation, boolean useDelay, boolean useSquaredInputs) {
+
+  }
```
Comment (Dev-Osmium) [inline src/main/java/frc/robot/subsystems/drivetrain/MockDriveTrain.java]: Might want to add printlns here just so we know it's running

File: src/main/java/frc/robot/subsystems/drivetrain/MockDriveTrain.java [hunk 2]
```diff
@@ -33,4 +33,13 @@ public double getEncoderPositionInches() {
...
+
+  public void moveUsingGyro(double forwardBackward, double rotation, boolean useDelay, boolean useSquaredInputs,
+      double heading) {
+
+  }
```
Comment (Dev-Osmium) [inline src/main/java/frc/robot/subsystems/drivetrain/MockDriveTrain.java]: same here

File: src/main/java/frc/robot/subsystems/drivetrain/RealDriveTrain.java [hunk 3]
```diff
@@ -22,6 +23,17 @@
...
+  // Gyro variables
+  private NavXGyroSensor navX;
+  private DriveTrain driveTrain;
+  private double savedAngle = 0;
+  private double newRotateValue = 0;
+  private boolean gyroCorrect = false;
+  private double currentDelay = 0;
+  private static final double kDelay = 25;
+  private static final double kProportion = 0.05;
+  private boolean invertGyroCorrect = false;
```
Comment (Dev-Osmium) [inline src/main/java/frc/robot/subsystems/drivetrain/RealDriveTrain.java]: I spy magic numbers that could be in a config

File: src/main/java/frc/robot/subsystems/drivetrain/MockDriveTrain.java (hunk 1 above)
Comment (JuliaLupaczyk) [reply src/main/java/frc/robot/subsystems/drivetrain/MockDriveTrain.java]: No.

File: src/main/deploy/robotConfigs/TheDroidYoureLookingFor/drivetrain.conf [hunk 4]
```diff
@@ -7,3 +7,8 @@ ports {
 drivetrain{
     ticksPerInch=1
 }
+gyrocorrect {
+    kdelay=25
+    kproportion=0.05
+    invertgyrocorrect=false
+}
```
Comment (Dev-Osmium) [inline src/main/deploy/robotConfigs/TheDroidYoureLookingFor/drivetrain.conf]: Good on you for adding them to the config

File: src/main/java/frc/robot/subsystems/drivetrain/RealDriveTrain.java (hunk 3 above)
Comment (Dev-Osmium) [inline src/main/java/frc/robot/subsystems/drivetrain/RealDriveTrain.java]: but you need to use them in the code

File: src/main/java/frc/robot/subsystems/drivetrain/RealDriveTrain.java [hunk 5]
```diff
@@ -22,6 +23,17 @@
...
 
+  // Gyro variables
+  private NavXGyroSensor navX;
+  private DriveTrain driveTrain;
+  private double savedAngle = 0;
+  private double newRotateValue = 0;
+  private boolean gyroCorrect = false;
+  private double currentDelay = 0;
+  private static final double kDelay = 25;
+  private static final double kProportion = 0.05;
```
Comment (themccubbins) [inline src/main/java/frc/robot/subsystems/drivetrain/RealDriveTrain.java]: Load magic numbers from the config ?

**Pull request merged**